### DISCUSSION TITLE
Server not automatically starting.

### DISCUSSION THREAD

**Discussion by ghost** [Ylianst/MeshCentral#5044] [Q&A]
Fresh install on Digital Ocean. Ubuntu 20.04

The only way I can get my server to start is to manually run `node ./node_modules/meshcentral --cert mesh.pc-support.uk.com`

`node -v = v18.14.0`
`npm -v = 9.6.2`

meshcentral.service:
```
[Unit]
Description=MeshCentral Server

[Service]
Type=simple
LimitNOFILE=1000000
# ExecStart=/usr/bin/node /home/pc-support/.nvm/versions/node/v18.14.0/bin/node
ExecStart=/home/pc-support/.nvm/versions/node/v18.14.0/bin/node
WorkingDirectory=/home/pc-support
Environment=NODE_ENV=production
User=pc-support
Group=pc-support
Restart=always
# Restart service after 10 seconds if node service crashes
RestartSec=10
# Set port permissions capability
AmbientCapabilities=cap_net_bind_service

[Install]
WantedBy=multi-user.target
```

Any help greatly appreciated.

**Comment by ghost** [comment]
vv

**Reply by ouddorp** [reply]
I'm using Pm2 for that: https://pm2.keymetrics.io/

**Reply by SomeGuru** [reply]
@ouddorp I just started exploring PM2 on my AWS instances, how are you liking the configuration this way, and secondly how has your reliability been since making the change?

I am not running in Docker, I want native settings to Ubuntu Server 22.04 vs. docker ran pm2 and mesh.

Ideally I am running mesh on 443:4433 and uptime-kuma on 6667 for monitoring all my domains.

I am currently exploring the network-online.target to get PM2 to launch after my network is stabilized on AWS as I have seen it sometimes launches before it becomes stable.

Ideally, I want to eventually run through a elb process with multiple server nodes that report back on the main and replicate the datasets with MongoDB or similar DB given its not yet another cloud service needed.

-SomeGuru

**Reply by ouddorp** [reply]
@SomeGuru Our setup is on AWS as well. We're running MC on a Debian t3a Ec2 instance, database is RDS Postgres (db.t3.micro)  and an application load balancer in front of it. I did choose Postgres and the application load balancer because we use it for other services as well. MC is very reliable, 454 devices we have currently, I'm only restarting MC in case of an update.

**Comment by muelv** [comment]
Hi.I do have the same problem, but using Windows Server as MeshCentral-Host. I already googled and found an answer of Ylianst, that you have to run meshcentral.exe using --install.

My Problem: I have a batch file, which is starting my MeshCentral:

@echo off
c:
cd \meshcentral
node node_modules/meshcentral --cert mcentral.my-domain.de

But after having used "meshcentral --install" my Service starts, but the WebService is not running. What can I do?

**Reply by SomeGuru** [reply]
@muelv What ports are open on your Windows Firewall settings? 

SomeGuru

**Comment by muelv** [comment]
Hi, My problem is solved, I had to type

node node_modules/meshcentral --cert my.domain.de

and now the service is running. Thank you very much.

Sven

 

 

Von: SomeGuru ***@***.***> 
Gesendet: Dienstag, 25. Juli 2023 22:02
An: Ylianst/MeshCentral ***@***.***>
Cc: muelv ***@***.***>; Mention ***@***.***>
Betreff: Re: [Ylianst/MeshCentral] Server not automatically starting. (Discussion #5044)

 

@muelv <https://github.com/muelv>  What ports are open on your Windows Firewall settings?

SomeGuru
